src/main/java/com/ea/repositories/GameReportRepository.java

```diff
@@ -15,17 +16,65 @@
 @Repository
 public interface GameReportRepository extends JpaRepository<GameReportEntity, Long> {
 
-    @Query("SELECT pc.address FROM GameReportEntity gr JOIN gr.personaConnection pc WHERE gr.game.id = :gameId AND gr.isHost = true AND pc.isHost = true AND pc.endTime IS NULL")
+    @Query("SELECT pc.address FROM GameReportEntity gr JOIN gr.personaConnection pc " +
+            "WHERE gr.game.id = :gameId AND gr.isHost = true AND pc.isHost = true AND pc.endTime IS NULL")
     List<String> findHostAddressByGameId(@Param("gameId") Long gameId);
 
     Optional<GameReportEntity> findByPersonaConnectionIdAndEndTimeIsNull(Long personaConnectionId);
 
     List<GameReportEntity> findByGameIdAndEndTimeIsNull(Long gameId);
 
-    List<GameReportEntity> findByPersonaConnectionPersonaPersAndGameStartTimeAndPlayTimeAndIsHostFalse(String pers, LocalDateTime startTime, int playTime);
+    List<GameReportEntity> findByPersonaConnectionPersonaPersAndGameStartTimeAndPlayTimeAndIsHostFalse(
+            String pers,
+            LocalDateTime startTime,
+            int playTime
+    );
 
     @Transactional
     @Modifying
     @Query("UPDATE GameReportEntity gr SET gr.endTime = :endTime WHERE gr.endTime IS NULL")
     int setEndTimeForAllUnfinishedGameReports(@Param("endTime") LocalDateTime endTime);
+
+    @Query("SELECT COUNT(DISTINCT gr.personaConnection.id) FROM GameReportEntity gr " +
+            "WHERE gr.endTime IS NULL AND gr.isHost = false")
+    int countActiveNonHostPlayers();
+
+    @Query("SELECT gr.personaConnection.persona.pers FROM GameReportEntity gr " +
+            "WHERE gr.game.id = :gameId AND gr.isHost = true AND gr.endTime IS NULL")
+    String findHostNameByGameId(@Param("gameId") Long gameId);
+
+    @Query("SELECT new com.ea.frontend.DTO$PlayerInfoDTO(" +
+            "gr.personaConnection.persona.pers, " +
+            "gr.isHost, " +
+            "gr.startTime) " +
+            "FROM GameReportEntity gr " +
+            "WHERE gr.game.id = :gameId " +
+            "AND gr.endTime IS NULL " +
+            "AND gr.isHost = false")
+    List<DTO.PlayerInfoDTO> findActivePlayersByGameId(@Param("gameId") Long gameId);
+
+    @Query("""
+        SELECT new com.ea.frontend.DTO$GameStatusDTO(
+            g.id,
+            g.name,
+            g.vers,
+            g.startTime,
+            g.maxsize,
+            h.personaConnection.persona.pers,
+            COUNT(p)
+        )
+        FROM GameEntity g
+        LEFT JOIN GameReportEntity h ON h.game = g AND h.isHost = true AND h.endTime IS NULL
+        LEFT JOIN GameReportEntity p ON p.game = g AND p.isHost = false AND p.endTime IS NULL
+        WHERE g.endTime IS NULL
+        GROUP BY g.id, g.name, g.vers, g.startTime, g.maxsize, h.personaConnection.persona.pers
+    """)
```

Comment:
Keep the code harmonized : use either """ or + to write queries on multiple lines, but use only one and stick to it